### PR TITLE
http in development mode crashed -

### PR DESCRIPTION
**Pull request by BaronGreenback**
Moved  appHost.EnableHttps to encompass certificate checking in dev mode

### CODE REVIEW

**Comment by mark-monteiro** [comment]
I'm not sure this will work because `EnableHttps` is false if `Certificate` is null. So you won't be able to run HTTPS in development mode without specifying a certificate, which kind of defeats the purpose of the development certificate.

What I think you also need to do is add an additional check for development mode to `SupportsHttps` in ApplicationHost.cs. But the network settings are kind of confusing so you'll need to double check that

**Comment by mark-monteiro** [comment]
@BaronGreenback Your latest commit looks really interesting! I think you should submit that as a separate PR though, it seems unrelated to the original purpose of this PR dealing with the ASP development certificate.

**Comment by BaronGreenback** [comment]
would love to, if i knew how! (Haven't worked with GitHub before)

I used visual studio - and pushed it to my area, but it merged PR with the last one.

How do i under that, and re-upload as a separate PR?

**Comment by mark-monteiro** [comment]
@barronpm If you're having trouble sorting out the changes with git I'd recommend reaching out for assistance on the `jellyfin-dev` matrix channel! It's usually pretty active I'm sure someone would be able to help out: https://jellyfin.org/docs/general/getting-help.html

File: Jellyfin.Server/Program.cs [hunk 1]
```diff
@@ -283,11 +282,18 @@ private static async Task StartApp(StartupOptions options)
...
-                                    listenOptions.UseHttps();
-                                    listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
-                                });
+                                    options.Listen(address, appHost.HttpsPort, listenOptions =>
+                                    {
+                                        listenOptions.UseHttps();
+                                        listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
+                                    });
+                                }
+                                catch (Exception ex)
```
Comment (mark-monteiro) [inline Jellyfin.Server/Program.cs:293]: This should catch the specific exception that is thrown if a default certificate is not configured

File: Jellyfin.Server/Program.cs [hunk 2]
```diff
@@ -283,11 +282,18 @@ private static async Task StartApp(StartupOptions options)
...
+                                {
+                                    _logger.LogError(ex, "Error whilst listing to https - Is a development certificate installed?");
```
Comment (mark-monteiro) [inline Jellyfin.Server/Program.cs:295]:
```suggestion
                                    _logger.LogError(ex, "Failed to listen to HTTPS using the ASP.NET Core HTTPS development certificate. Please ensure it has been installed and set as trusted.");
```

File: Jellyfin.Server/Program.cs [hunk 3]
```diff
@@ -306,11 +312,18 @@ private static async Task StartApp(StartupOptions options)
...
-                            });
+                                options.ListenAnyIP(appHost.HttpsPort, listenOptions =>
+                                {
+                                    listenOptions.UseHttps();
+                                    listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
+                                });
+                            }
+                            catch (Exception ex)
+                            {
+                                _logger.LogError(ex, "Error whilst listing to https - Is a development certificate installed?");
```
Comment (mark-monteiro) [inline Jellyfin.Server/Program.cs:325]:
```suggestion
                                _logger.LogError(ex, "Failed to listen to HTTPS using the ASP.NET Core HTTPS development certificate. Please ensure it has been installed and set as trusted");
```

**Pull request merged**